### ISSUE TITLE
query_timeout limits whole transaction time

### ISSUE THREAD

**Issue by vincentcr** [pgbouncer/pgbouncer#579]
`query_timeout` closes any connection with a transaction older than query_timeout.

Seems like the same issue as https://github.com/pgbouncer/pgbouncer/issues/22 which was meant to be fixed 2 years ago.

Steps to repro:

```
> BEGIN;
> /* wait query_timeout seconds */
> SELECT 1;
ERROR:  query timeout
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
```

Tested with 1.15.0 and 1.12.0.

**Comment by jcoleman** [comment]
I don't think this is describing the same thing as #22. Rather than the whole transaction being limited by `query_timeout` (I've confirmed I can't reproduced that) what you're actually noting is that `query_timeout` is effectively acting like `idle_transaction_timeout`, which is IMO still unexpected.
